site/src/pages/DeploySettingsPage/NotificationsPage/NotificationsPage.tsx

```diff
@@ -0,0 +1,269 @@
+            </ToggleButton>
+          </Tooltip>
+        );
+      })}
+    </ToggleButtonGroup>
+  );
+};
+
+export const NotificationsPage: FC = () => {
+  const [searchParams] = useSearchParams();
+  const { deploymentValues } = useDeploySettings();
+  const [templatesByGroup, dispatchMethods] = useQueries({
+    queries: [
+      {
+        ...systemNotificationTemplates(),
+        select: selectTemplatesByGroup,
+      },
+      notificationDispatchMethods(),
+    ],
+  });
+  const ready =
+    templatesByGroup.data && dispatchMethods.data && deploymentValues;
+  const tab = searchParams.get("tab") || "events";
+
+  return (
+    <>
+      <Helmet>
+        <title>{pageTitle("Notifications Settings")}</title>
+      </Helmet>
+      <Section
+        title="Notifications"
+        description={
+          <>
+            Control delivery methods for notifications on this deployment.
+            Notifications may be disabled in your{" "}
+            <Link to="/settings/notifications">profile settings</Link>.
+          </>
+        }
+        layout="fluid"
+      >
+        <Tabs active={tab}>
+          <TabsList>
+            <TabLink to="?tab=events" value="events">
+              Events
+            </TabLink>
+            <TabLink to="?tab=settings" value="settings">
+              Settings
+            </TabLink>
+          </TabsList>
+        </Tabs>
+
+        <div css={styles.content}>
+          {ready ? (
+            tab === "events" ? (
+              <EventsView
+                defaultMethod={castNotificationMethod(
+                  dispatchMethods.data.default,
+                )}
+                availableMethods={dispatchMethods.data.available.map(
+                  castNotificationMethod,
+                )}
+                notificationsConfig={deploymentValues.config.notifications}
+                templatesByGroup={templatesByGroup.data}
+              />
+            ) : (
+              <OptionsTable
+                options={deploymentValues?.options.filter((o) =>
+                  deploymentGroupHasParent(o.group, "Notifications"),
+                )}
+              />
+            )
+          ) : (
+            <Loader />
+          )}
+        </div>
+      </Section>
+    </>
+  );
+};
+
+type EventsViewProps = {
+  defaultMethod: NotificationMethod;
+  availableMethods: NotificationMethod[];
+  notificationsConfig?: NotificationsConfig;
+  templatesByGroup: ReturnType<typeof selectTemplatesByGroup>;
+};
+
+const EventsView: FC<EventsViewProps> = ({
+  defaultMethod,
+  availableMethods,
+  notificationsConfig,
+  templatesByGroup,
+}) => {
+  const isUsingWebhook = availableMethods.includes("webhook");
+  const webhookEndpoint = notificationsConfig?.webhook.endpoint;
+
+  return (
+    <Stack spacing={3}>
+      {isUsingWebhook && !webhookEndpoint && (
+        <Alert severity="warning">
+          Webhook method is enabled, but the endpoint is not configured.
+        </Alert>
+      )}
```

Comment:
We can't assume SMTP will always be configured. We should have the same check here.